201-vm-customimage-new-storageaccount/azuredeploy.json

```diff
@@ -0,0 +1,307 @@
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-05-01-preview",
+      "dependsOn": [
+        "[concat('Microsoft.Storage/storageAccounts/', variables('storageAccountName'))]",
+        "[concat('Microsoft.Network/networkInterfaces/', variables('vmNames')[0],'NIC')]"
+      ],
+      "tags": {
+        "displayName": "[variables('vmNames')[0]]"
+      },
+      "properties": {
+        "hardwareProfile": {
+          "vmSize": "[variables('vmSizes')[0]]"
+        },
+        "osProfile": {
+          "computerName": "[variables('vmNames')[0]]",
+          "adminUsername": "[parameters('AdminUsername')]",
+          "adminPassword": "[parameters('AdminPassword')]"
+        },
+        "storageProfile": {
+          "imageReference": {
+            "publisher": "[variables('imagePublisher')]",
+            "offer": "[variables('imageOffer')]",
+            "sku": "[variables('windowsOSVersion')]",
+            "version": "latest"
+          },
+          "osDisk": {
+            "name": "[concat(variables('vmNames')[0],'-osdisk')]",
+            "vhd": {
+              "uri": "[concat('http://', variables('storageAccountName'), '.blob.core.windows.net/', variables('vhdStorageAccountContainerName'), '/',variables('vmNames')[0],'-osdisk' , '.vhd')]"
+            },
+            "caching": "ReadWrite",
+            "createOption": "FromImage"
+          }
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('nicNames')[0]))]"
+            }
+          ]
+        }
+      },
+      "resources": [
+        {
+          "comments": "Custom Script that copies VHDs from source storage account to destination storage account",
+          "apiVersion": "2015-05-01-preview",
+          "type": "extensions",
+          "name": "[concat(variables('VMNames')[0],'CustomScriptExtension')]",
+          "location": "[resourceGroup().location]",
+          "dependsOn": [
+            "[concat('Microsoft.Compute/virtualMachines/', variables('vmNames')[0])]"
+          ],
+          "properties": {
+            "publisher": "Microsoft.Compute",
+            "type": "CustomScriptExtension",
+            "typeHandlerVersion": "1.4",
+            "settings": {
+              "fileUris": [
+                  "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-customimage-new-storageaccount/ImageTransfer.ps1"
+              ],
+              "commandToExecute": "[concat('powershell -ExecutionPolicy Unrestricted -File ','ImageTransfer.ps1 -SourceImage ',parameters('sourceImages'),' -SourceSAKey ', parameters('sourceStorageAccountKey'), ' -DestinationURI https://', variables('StorageAccountName'), '.blob.core.windows.net/vhds', ' -DestinationSAKey ', listKeys(concat('Microsoft.Storage/storageAccounts/', variables('StorageAccountName')), '2015-05-01-preview').key1 )]"
```

Comment:
Since the storage account keys are being passed in to the script, the commandToExecute should be in the protected settings instead